auth/proxy.go

```diff
@@ -22,7 +22,12 @@ type ProxyAuth struct {
 func (a ProxyAuth) Auth(r *http.Request, usr users.Store, setting *settings.Settings, srv *settings.Server) (*users.User, error) {
 	username := r.Header.Get(a.Header)
 	user, err := usr.Get(srv.Root, username)
-	if errors.Is(err, fbErrors.ErrNotExist) {
+	if err != nil {
+		// Return the error unless it is a user does not exist error
+		if !errors.Is(err, fbErrors.ErrNotExist) {
+			return nil, err
+		}
+
```

Comment:
This code is hard to read. I think it would be better to extract user creation into the createUser function
```go
if errors.Is(err, fbErrors.ErrNotExist) {
    return createUser()
}
return user, err
```